app/model/commands/PublishAtomCommand.scala

```diff
@@ -45,44 +45,66 @@ case class PublishAtomCommand(
       AtomPublishFailed("Atom status set to private")
     }
 
-    getActiveAsset(previewAtom) match {
-      case Some(asset) if asset.platform == Youtube =>
-        val duration = youtube.getDuration(asset.id)
+    val contentChangeDetails = thriftPreviewAtom.contentChangeDetails
+    val now = Instant.now().toEpochMilli
 
-        val blockAds = previewAtom.category match {
-          case Category.Hosted | Category.Paid => {
-            true
-          }
-          case _ => {
-            if (duration.getOrElse(0L) < youtube.minDurationForAds) true else previewAtom.blockAds
-          }
-        }
-
-        val privacyStatus: Future[PrivacyStatus] = (previewAtom.channelId, previewAtom.privacyStatus) match {
-          case (Some(channel), Some(status)) if youtube.unlistedWithoutPermissionChannels.contains(channel) && status == PrivacyStatus.Public => {
-            permissions.getStatusPermissions(user.email).map(permissions => {
-              val hasMakePublicPermission = permissions.setVideosOnAllChannelsPublic
-
-              if (hasMakePublicPermission){
-                PrivacyStatus.Public
-              } else {
-                log.info(s"User ${user.email} does not have permission to publish atom ${previewAtom.id} as Public, setting as Unlisted")
-                PrivacyStatus.Unlisted
-              }
+    (contentChangeDetails.scheduledLaunch, contentChangeDetails.embargo) match {
+      case (_, Some(embargo)) => {
```

Comment:
Shouldn't this also check ` if embargo.date > now`? Currently the embargo date value/label do not get automatically removed when the embargo date is reached

Reply:
good spot! changed.